### ISSUE TITLE
Image resize not working for some images (Picasso 2.5.2 and okhttp 2.3, okio 1.3)

### ISSUE THREAD

**Issue by gkishor** [square/picasso#987]
Thanks for this awesome library.

On a few occasions, I have noticed that Picasso is not loading images when resize is applied.

Following is an example:

		Picasso.with(this)
				.load("http://myntra.myntassets.com/images/style/properties/Silk-Bazar-Set-of-5-Chiffon-Fashion-Sarees_0afbb54ed6889434b8e11341d9b0132e_images.jpg")				
				.resize(0, 166)
				.into(imgView);

The same image is loaded properly without the resize operation:
		Picasso.with(this)
				.load("http://myntra.myntassets.com/images/style/properties/Silk-Bazar-Set-of-5-Chiffon-Fashion-Sarees_0afbb54ed6889434b8e11341d9b0132e_images.jpg")				
				.into(imgView);

Following is the logcat output with logging enabled:

04-21 16:56:45.937: D/Picasso(4433): Main        created      [R1] Request{http://myntra.myntassets.com/images/style/properties/Silk-Bazar-Set-of-5-Chiffon-Fashion-Sarees_0afbb54ed6889434b8e11341d9b0132e_images.jpg}
04-21 16:56:45.947: D/Picasso(4433): Dispatcher  enqueued     [R1]+5ms 
04-21 16:56:45.947: D/Picasso(4433): Hunter      executing    [R1]+7ms 
04-21 16:56:46.047: W/EGL_emulation(4433): eglSurfaceAttrib not implemented
04-21 16:56:46.157: D/dalvikvm(4433): GC_FOR_ALLOC freed 355K, 34% free 2980K/4480K, paused 5ms, total 17ms
04-21 16:56:47.097: D/Picasso(4433): Dispatcher  retrying     [R1]+1151ms 
04-21 16:56:47.097: D/Picasso(4433): Hunter      executing    [R1]+1152ms 
04-21 16:56:47.177: D/dalvikvm(4433): GC_FOR_ALLOC freed 421K, 35% free 2972K/4572K, paused 4ms, total 4ms
04-21 16:56:48.047: D/Picasso(4433): Dispatcher  retrying     [R1]+2102ms 
04-21 16:56:48.047: D/Picasso(4433): Hunter      executing    [R1]+2103ms 
04-21 16:56:48.047: D/Picasso(4433): Dispatcher  batched      [R1]+2104ms for error
04-21 16:56:48.257: D/Picasso(4433): Dispatcher  delivered    [R1]+2311ms 
04-21 16:56:48.257: D/Picasso(4433): Main        errored      [R1]+2312ms 

I have run a wireshark trace and identified that the http request to load the image is succeeding and the server did return a HTTP 200 and ~500K payload with the jpeg data. So, it is not a network issue. Further the image is not getting loaded only if resize is specified. If the resize is removed, everything works fine.

**Comment by dnkoutso** [comment]
Dup of #907 

```
8678-8678/com.example.picasso W/System.err﹕ java.io.IOException: Cannot reset
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at com.squareup.picasso.MarkableInputStream.reset(MarkableInputStream.java:99)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at com.squareup.picasso.BitmapHunter.decodeStream(BitmapHunter.java:140)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at com.squareup.picasso.BitmapHunter.hunt(BitmapHunter.java:217)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at com.squareup.picasso.BitmapHunter.run(BitmapHunter.java:159)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at java.util.concurrent.FutureTask.run(FutureTask.java:237)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at java.lang.Thread.run(Thread.java:818)
05-09 14:43:46.259    8678-8678/com.example.picasso W/System.err﹕ at com.squareup.picasso.Utils$PicassoThread.run(Utils.java:411)
```

Will be fixed in 2.5.3.

**Comment by dnkoutso** [comment]
Update `BitmapHunter` 

```java
    long mark = markStream.savePosition(65536); 
```

Higher number would process the image.
